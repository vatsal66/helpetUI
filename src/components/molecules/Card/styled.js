import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  background-color: ${({ theme }) => theme.colors.background};
  border-radius: 25px;
  padding: 30px 35px;
  min-height: 280px;
  max-width: 350px;
  z-index: 100;

  transition: transform 150ms ease;
  transform: ${({ animated }) => (animated ? 'scale(1)' : 'scale(0)')};

  & > * + * {
    margin-top: 10px;
  }

  @media (max-width: 374px) {
    max-width: 300px;
  }
`;

export const TypeText = styled.p`
  font-size: 1em;
  color: ${({ theme }) => theme.colors.description};
  text-transform: uppercase;
`;

export const Title = styled.h2`
  font-size: 36px;
  font-weight: 700;
  letter-spacing: -2px;
  line-height: 1;
  text-align: center;
`;

export const Description = styled.h3`
  color: ${({ theme }) => theme.colors.description};
  text-align: center;
  font-size: 1.2em;
  font-weight: normal;
`;
