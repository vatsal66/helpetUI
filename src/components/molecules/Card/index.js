import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import scrollTo from 'gatsby-plugin-smoothscroll';

import { Button } from '@components/atoms';

import CardDetail from '@assets/icons/card-detail.svg';

import { TypeText, Container, Title, Description } from './styled';

const Card = ({ className, animated, card }, ref) => {
  const { messages } = useIntl();

  return (
    <Container {...{ ref, className, animated }}>
      <TypeText>{messages[`cards.${card}.type`]}</TypeText>
      <CardDetail width={140} />
      <Title>{messages[`cards.${card}.title`]}</Title>
      <Description>{messages[`cards.${card}.description`]}</Description>
      <Button
        as="a"
        padding={[12, 24]}
        onClick={() => scrollTo('#header-download')}
      >
        {messages['downloadApp']}
      </Button>
    </Container>
  );
};

Card.propTypes = {
  card: PropTypes.string.isRequired,
  className: PropTypes.string,
  animated: PropTypes.bool,
};

Card.defaultProps = {
  className: '',
  animated: false,
};

export default forwardRef(Card);
