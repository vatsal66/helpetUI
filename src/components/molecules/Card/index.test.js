import React from 'react';

import renderWithTheme from '@mocks/renderWithTheme';

import Card from './index';

describe('<Card />', () => {
  const props = {
    messageTitleId: 'petId.title',
    messageDescriptionId: 'petId.description',
  };

  it('should match snapshot file', () => {
    const { asFragment } = renderWithTheme(<Card {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
