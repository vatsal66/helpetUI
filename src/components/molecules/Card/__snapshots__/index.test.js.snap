// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snapshot file 1`] = `
<DocumentFragment>
  <div
    class="sc-AxheI dLGpGh"
  >
    <p
      class="sc-Axmtr lkqzQX"
    >
      Documentazione
    </p>
    <test-file-stub
      width="140"
    />
    <h1
      class="sc-AxmLO dheAqI"
    >
      Pet ID e Passaporto
    </h1>
    <h2
      class="sc-fzozJi kNXHwV"
    >
      Tutta la documentazione del tuo miglior amico racchiusa all’interno di un’unica app
    </h2>
    <button
      class="sc-AxjAm ALgEm"
    >
      Download App
    </button>
  </div>
</DocumentFragment>
`;
